src/array-repeat-strategy.js

```diff
@@ -20,16 +20,79 @@ export class ArrayRepeatStrategy {
   * @param items The new array instance.
   */
   instanceChanged(repeat, items) {
-    if (repeat.viewsRequireLifecycle) {
-      let removePromise = repeat.removeAllViews(true);
-      if (removePromise instanceof Promise) {
-        removePromise.then(() => this._standardProcessInstanceChanged(repeat, items));
+    // if the new instance does not contain any items, 
+    // just remove all views and don't do any further processing
+    if (!items || items.length === 0) {
+      repeat.removeAllViews(true);
+      return;
```

Comment:
I just noticed this chunk wasn't necessary anymore (no need to process, because there are no views to add):
```js
      if (removePromise instanceof Promise) {
        removePromise.then(() => this._standardProcessInstanceChanged(repeat, items));
      }
```
So I removed that too, making the code even cleaner.